lib/perl/Genome/Site/TGI/Extension/UnarchiveInstrumentData.pm

```diff
@@ -0,0 +1,209 @@
+
+    my $dv = $self->volume;
+    if ($instrument_data->class ne 'Genome::InstrumentData::Solexa') { return 1; }
+
+    my $bam_path = $instrument_data->bam_path;
+    if (defined($bam_path)) {
+        if (-e $bam_path) {
+            return 1;
+        }
+    } else {
+        if (defined($instrument_data->archive_path)) {
+            $self->warning_message('Skipping old data with archive_path: '. $instrument_data->__display_name__);
+            return 1;
+        }
+    }
+
+    $self->status_message('Working on missing BAM %s for instrument data %s', $bam_path, $instrument_data->id);
+
+    my ($bam_filename, $bam_dirname) = File::Basename::fileparse($bam_path);
+    $bam_dirname =~ s/\/$//;
+    my @bam_dirnames = File::Spec->splitdir($bam_dirname);
+
+    my ($unarchived_bam_path) = glob(File::Spec->join($dv->mount_path,'*','csf_*',$bam_filename));
+    if (!defined($unarchived_bam_path)) {
+        # Try one more tine allowing for an additional directory in the path, ie. "condensed"
+        ($unarchived_bam_path) = glob(File::Spec->join($dv->mount_path,'*','*','csf_*',$bam_filename));
+    }
+    unless ($unarchived_bam_path) {
+        Genome::Sys->shellcmd(
+            cmd => [$lims_unarchive_script_path,$instrument_data->id,$dv->mount_path],
+        );
+    } else {
+        my $allocation = $self->get_or_create_allocation($instrument_data);
+
+        my ($unarchived_bam_filename,$unarchived_bam_dirname) = File::Basename::fileparse($unarchived_bam_path);
+        $unarchived_bam_dirname =~ s/\/$//;
+        my $allocation_bam_dirname = File::Spec->join($allocation->absolute_path,$bam_dirnames[-1]);
+        if (-e $allocation_bam_dirname) {
+            $self->warning_message('Already found allocated path, skipping: %s', $allocation_bam_dirname);
+            $self->warning_message('Please remove the unarchived path: %s', $unarchived_bam_dirname);
+            return;
+        }
+
+        unless ($self->validate_bam_is_complete($unarchived_bam_path,$instrument_data)) {
+            $self->status_message('It appears the unarchiving is still in progress for: %s', $unarchived_bam_path);
+            return;
+        }
+
+        $self->debug_message('Rsync contents of %s to %s ', $unarchived_bam_dirname, $allocation_bam_dirname);
+        unless (
+            Genome::Sys->rsync_directory(
+                source_directory => $unarchived_bam_dirname,
+                target_directory => $allocation_bam_dirname,
+            )
+          ) {
+            $self->fatal_message('Failed to rsync from %s to %s', $unarchived_bam_dirname, $allocation_bam_dirname);
+        }
+        unless ($ENV{UR_DBI_NO_COMMIT}) {
+            Genome::Sys::CommitAction->create(
+                on_commit => sub {
+                    unless(Genome::Sys->remove_directory_tree($unarchived_bam_dirname)) {
+                        $self->fatal_message('Failed to remove directory: %s', $unarchived_bam_dirname);
```

Comment:
Should this be a warning_message() instead of fatal_message().  You don't want the directory to be removed until commit, but I don't think we want the commit to fail because the directory isn't removed.  The rsync worked, so the copy in the allocation should be fine.  One reason this could happen is due to permissions on the unarchived files/directories.  Currently this requires root access to modify.  If the file was removed, but not the directory... and the commit does not save, then the unarchive has to be rerun and permissions modified PROactively rather than retroactively.  The downside is that directories and/or files could be left around unknowingly.